publisher/publisher.go

```diff
@@ -208,10 +213,23 @@ func (p *Publisher) Publish(ctx context.Context, msg interface{}, opts ...Option
 
 func ensureTopic(ctx context.Context, name string, namespace *servicebus.Namespace, opts ...servicebus.TopicManagementOption) (*servicebus.TopicEntity, error) {
 	tm := namespace.NewTopicManager()
-	te, err := tm.Get(ctx, name)
-	if err == nil {
+	ensure := func() (interface{}, error) {
+		te, err := tm.Get(ctx, name)
+		if err == nil {
+			return te, nil
+		}
+		te, err = tm.Put(ctx, name, opts...)
+		if err != nil {
+			// let all errors be retryable for now. application only hit this once on topic creation.
+			return nil, common.Retryable(err.Error())
+		}
 		return te, nil
 	}
+	entity, err := common.Retry(5, 1*time.Second, ensure)
```

Comment:
max 5 sec toll on service start-up seems reasonable.
test shows that it passes on 2nd attempt, so could reduce this further.

Reply:
You could explictly re-get on 409 and skip the retry all together assuming that callers of ensureTopic crash the program and it will just get restarted by k8s other orchestrator.

Reply:
yes, but crashing the program is not working out for e2e tests, cause every runs was causeing the replicas to crash at least once.